### PR TITLE
Adding documentation for certificate renewal

### PR DESCRIPTION
**Pull request by pravinpushkar**
Signed-off-by: Pravin Pushkar <ppushkar@microsoft.com>

Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [na] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [na] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

### CODE REVIEW

**Comment by msfussell** [comment]
Reviewing this now, and will have to also include this change in this PR https://github.com/dapr/cli/issues/893

File: daprdocs/content/en/reference/cli/dapr-mtls/_index.md [hunk 1]
```diff
@@ -30,8 +30,9 @@ dapr mtls [command]
 ### Available Commands
 
 ```txt
-expiry      Checks the expiry of the root certificate
-export      Export the root CA, issuer cert and key from Kubernetes to local files
+expiry              Checks the expiry of the root certificate
+export              Export the root CA, issuer cert and key from Kubernetes to local files
```
Comment (msfussell) [inline daprdocs/content/en/reference/cli/dapr-mtls/_index.md]: I notice that the wording on these CLI references is wrong since it says "for Kubernetes" in several places, when it should just say what it does and then then supported platforms tell you where this works.

So this sentence should be 
```suggestion
export              Export the root Certificate Authority (CA), issuer cert and issuer key to local files
```

Comment (msfussell) [reply daprdocs/content/en/reference/cli/dapr-mtls/_index.md]: Can you also apply this approach to the existing export and expiry commands in these files
https://docs.dapr.io/reference/cli/dapr-mtls/dapr-mtls-export/
https://docs.dapr.io/reference/cli/dapr-mtls/dapr-mtls-expiry/

For example in the export file 
1) Change "Export the root CA, issuer cert and key from Kubernetes to local files" to "Export the root Certificate Authority (CA), issuer cert and key to local files"

Comment (msfussell) [reply daprdocs/content/en/reference/cli/dapr-mtls/_index.md]: Actually I have just realized that these export, expiry and renew should all have -k as an option to say that these apply to Kubernetes. No single CLI command should be tied to a single platform environment, so this is a bug we need to fix. Especially when we claim/show this in the docs https://docs.dapr.io/concepts/security-concept/#mtls-in-self-hosted-mode and there are companies who would want to deploy the Sentry service on-premises. In other words it should be this.

dapr mtls export -k -o ./certs
dapr mtls expiry -k 
dapr mtls renew -k

in the same way we have
dapr mtls -k

File: daprdocs/content/en/reference/cli/dapr-mtls/_index.md [hunk 2]
```diff
@@ -30,8 +30,9 @@ dapr mtls [command]
 ### Available Commands
 
 ```txt
-expiry      Checks the expiry of the root certificate
-export      Export the root CA, issuer cert and key from Kubernetes to local files
+expiry              Checks the expiry of the root certificate
```
Comment (msfussell) [inline daprdocs/content/en/reference/cli/dapr-mtls/_index.md]:
```suggestion
expiry              Checks the expiry of the root Certificate Authority (CA) certificate
```

File: daprdocs/content/en/reference/cli/dapr-mtls/_index.md [hunk 3]
```diff
@@ -30,8 +30,9 @@ dapr mtls [command]
...
+export              Export the root CA, issuer cert and key from Kubernetes to local files
+renew-certificate   Rotates root CA, issuer cert and issuer key in Kubernetes secret used by Sentry service
```
Comment (msfussell) [inline daprdocs/content/en/reference/cli/dapr-mtls/_index.md]:
```suggestion
renew-certificate   Rotates the existing root Certificate Authority (CA), issuer cert and issuer key
```

File: daprdocs/content/en/reference/cli/dapr-mtls/_index.md [hunk 4]
```diff
@@ -40,6 +41,7 @@ You can learn more about each sub command from the links below.
 
 - [`dapr mtls expiry`]({{< ref dapr-mtls-expiry.md >}})
 - [`dapr mtls export`]({{< ref dapr-mtls-export.md >}})
+- [`dapr mtls renew-certificate`]({{< ref dapr-mtls-renew-certificate.md >}})
 
 ### Examples
```
Comment (msfussell) [inline daprdocs/content/en/reference/cli/dapr-mtls/_index.md:47]: Can you update the comment in the example below to say "for K8s" like this
```

# Check if mTLS is enabled on the Kubernetes cluster
dapr mtls -k
```

File: daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+```bash
+dapr mtls renew-certificate [flags]
+```
+
+### Flags
+
+| Name           | Environment Variable | Default           | Description                                 |
+| -------------- | -------------------- | ----------------- | ------------------------------------------- |
+| `--help`, `-h` |                      |                   | help for renew-certificate
+| `--kubernetes`, `-k` |                      | `false` | supprted platform|                             |
```
Comment (msfussell) [inline daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md:27]: Ah! Perfect. You added the -k command for this! Excellent. Now we just need to add these for the expiry and export commands

File: daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
+---
+type: docs
+title: "mtls renew certificate CLI command reference"
+linkTitle: "mtls renew certificate"
+description: "Detailed information on the mtls renew certificate CLI command"
+weight: 3000
+---
+
+### Description
+This command can be used to renew expiring Dapr certificates in Kubernetes cluster.
```
Comment (msfussell) [inline daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md]:
```suggestion
This command can be used to renew expiring Dapr certificates. For example the Dapr Sentry service can generate default root and issuer certificates used by applications. For more information see [secure Dapr to Dapr communication]({{< ref secure-dapr-to-dapr-communication >}})
```

File: daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+### Flags
+
+| Name           | Environment Variable | Default           | Description                                 |
+| -------------- | -------------------- | ----------------- | ------------------------------------------- |
+| `--help`, `-h` |                      |                   | help for renew-certificate
+| `--kubernetes`, `-k` |                      | `false` | supprted platform|                             |
+| `--valid-until`  |                      | 365 days |  Validity for newly created certificates |
+| `--restart`  |                      | false |  Restarts Dapr control plane services (Sentry service, Operator service and Placement server) |
+| `--timeout`  |                      | 300 sec |  The timeout for the certificate renewal process |
```
Comment (msfussell) [inline daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md:30]: I do not understand this timeout. This is the time for the renewal process to complete? What if this is a 300 node K8s cluster, will this succeed?

File: daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+| `--ca-root-certificate`  |                      |  |  User provided root certificate pem file path|
```
Comment (msfussell) [inline daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md]:
```suggestion
| `--ca-root-certificate`  |                      |  |  File path to user provided PEM root certificate
```

File: daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+| `--issuer-public-certificate`  |                      |  |  User provided issuer certificate pem file path|
```
Comment (msfussell) [inline daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md]:
```suggestion
| `--issuer-public-certificate`  |                      |  |  File path to user provided PEM issuer certificate |
```

File: daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+| `--issuer-private-key`  |                      |  |  User provided issue private key file path|
```
Comment (msfussell) [inline daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md]:
```suggestion
| `--issuer-private-key`  |                      |  |  File path to user provided PEM issue private key |
```

File: daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+| `--private-key`  |                      |  |  User provided root.key file which is used to generate root certificate|
+
+### Examples
+
+#### Renew certificates by generating fresh new certificates
```
Comment (msfussell) [inline daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md]:
```suggestion
#### Renew certificates by generating brand new certificates
```

File: daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+Generates new root and issuer certificates for Dapr kubernetes cluster with a default validity of 365 days.
```
Comment (msfussell) [inline daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md]:
```suggestion
Generates new root and issuer certificates for the Kubernetes cluster with a default validity of 365 days. The certificates are not applied to the Dapr control plane.
```

File: daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+```bash
+dapr mtls renew-certificate -k
+```
+Generates new root and issuer certificates for kubernetes cluster with a default validity of 365 days and restart the control plane services.
```
Comment (msfussell) [inline daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md]:
```suggestion
Generates new root and issuer certificates for the Kubernetes cluster with a default validity of 365 days and restarts the Dapr control plane services.
```

File: daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+```bash
+dapr mtls renew-certificate -k --restart
+```
+Generates new root and issuer certificates for kubernetes cluster with a given validity.
```
Comment (msfussell) [inline daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md]:
```suggestion
Generates new root and issuer certificates for the Kubernetes cluster with a given validity time.
```

File: daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+```bash
+dapr mtls renew-certificate -k --valid-until <no of days>
+```
+Generates new root and issuer certificates for kubernetes cluster with a given validity and restart the control place services.
```
Comment (msfussell) [inline daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md]:
```suggestion
Generates new root and issuer certificates for the Kubernetes cluster with a given validity time and restarts the Dapr control place services.
```

File: daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+```bash
+dapr mtls renew-certificate -k --valid-until <no of days> --restart
+```
+#### Renew certificate by using user provided certificates
+Rotates certificate of your kubernetes cluster with provided ca.pem, issuer.pem and issuer.key file path and restart the control plane services
```
Comment (msfussell) [inline daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md]:
```suggestion
Rotates certificates for the Kubernetes cluster with the provided ca.pem, issuer.pem and issuer.key file paths and restarts the Dapr control plane services
```

File: daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+```bash
+dapr mtls renew-certificate -k --ca-root-certificate <ca.pem> --issuer-private-key <issuer.key> --issuer-public-certificate <issuer.pem> --restart
+```
+Rotates certificate of your kubernetes cluster with provided ca.pem, issuer.pem and issuer.key file path.
```
Comment (msfussell) [inline daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md]:
```suggestion
Rotates certificates for the Kubernetes cluster with the provided ca.pem, issuer.pem and issuer.key file paths.
```

File: daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+```bash
+dapr mtls renew-certificate -k --ca-root-certificate <ca.pem> --issuer-private-key <issuer.key> --issuer-public-certificate <issuer.pem>
+```
+#### Renew certificates by generating fresh certificates using provided root private key
```
Comment (msfussell) [inline daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md]:
```suggestion
#### Renew certificates by generating brand new certificates using the provided root private key
```

File: daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+Uses existing private root.key to generate new root and issuer certificates for kubernetes cluster with a given validity for created certs.
```
Comment (msfussell) [inline daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md]:
```suggestion
Uses existing private root.key to generate new root and issuer certificates for the Kubernetes cluster with a given validity time for created certs.
```

File: daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md [hunk 20]
```diff
@@ -0,0 +1,73 @@
...
+```bash
+dapr mtls renew-certificate -k --private-key myprivatekey.key --valid-until <no of days>
+```
+Uses existing private root.key to generate new root and issuer certificates for kubernetes cluster with a default validity of 365 days for created certs.
```
Comment (msfussell) [inline daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md]:
```suggestion
Uses the existing private root.key to generate new root and issuer certificates for the Kubernetes cluster.
```

File: daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md (hunk 19 above)
Comment (msfussell) [reply daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md]: What this topic is lacking is any of the guidance on best practices. So next we need to update the guidance in this document https://docs.dapr.io/operations/security/mtls/ especially in the https://docs.dapr.io/operations/security/mtls/#updating-root-or-issuer-certs section. This section was recently updated before these commands were available. I suggest that you talk with Bernd who was the person responsible for testing this out. 

If effect we want to be clear that the user has two primary choices on guidance showing how to put all the mtls commands together and these choice are;
1) Use the self-signed Dapr generated root and issuer certs which is easy, but you will encountered downtime on renewal
2) Use this renew command with a private root key to update the deploy and renew root and issuer certs to avoid downtime. We should show the complete set of steps and commands to accomplish this. 

Bottom line, please work with me and Bernd to update this topic https://docs.dapr.io/operations/security/mtls/ and in particular this section https://docs.dapr.io/operations/security/mtls/#updating-root-or-issuer-certs. For example lets split this topic into Kubernetes and self hosted.

**Comment by msfussell** [comment]
Please fix this YouTube link https://docs.dapr.io/operations/security/mtls/#community-call-video-on-certificate-rotation. I also suggest that it was be really good to demo how to deploy new certificates and renew them at one of the April community calls. This is a very important topic to get people to understand for their cluster operation.

File: daprdocs/content/en/reference/cli/dapr-mtls/_index.md (hunk 1 above)
Comment (mukundansundar) [reply daprdocs/content/en/reference/cli/dapr-mtls/_index.md]: > No single CLI command should be tied to a single platform environment, so this is a bug we need to fix.

There are existing commands which are like that `dapr components -k`, `dapr configurations -k` etc ... (only k8s)
Please create an issue in CLI and this can triaged for future releases. IMO it is a new feature and not a bug ... since the way we query data for self-hosted mode is completely different. 

@msfussell

File: daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md (hunk 7 above)
Comment (pravinpushkar) [reply daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md:30]: This is actually default Helm wait timeout for any process like install/upgrade to complete. If any operation is supposed to take more time, we can set this flag to higher value to stop timing out helm operation. see `helm install/upgrade -h`

**Comment by pravinpushkar** [comment]
> Reviewing this now, and will have to also include this change in this PR [dapr/cli#893](https://github.com/dapr/cli/issues/893)

@msfussell  Since the warning message will be present with every commands in k8s mode, what could be the best place to put this information? Can we put this in the description section for each command where ever this is valid.

File: daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md (hunk 19 above)
Comment (pravinpushkar) [reply daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md]: @msfussell  As of now these command's implementation is done only for k8s mode. So in the k8s section of https://docs.dapr.io/operations/security/mtls/#updating-root-or-issuer-certs we can update the best practice along with cli command available for that.
I think we can create a separate PR for updating this ? Kindly share your thoughts.
@berndverst

**Comment by msfussell** [comment]
> > Reviewing this now, and will have to also include this change in this PR [dapr/cli#893](https://github.com/dapr/cli/issues/893)
> 
> @msfussell Since the warning message will be present with every commands in k8s mode, what could be the best place to put this information? Can we put this in the description section for each command where ever this is valid.

Yes, you should include this message with each of the respective CLI commands that this affects. I would add a message section, to each CLI command. Below is my MD format suggestion



### Warning messages
This command can issue warning messages.

#### Root certificate renewal warning
If the mtls root certificate deployed to the Kubernetes cluster expires in under 30 days the following warning message is displayed:

```bash
Dapr root certificate of your Kubernetes cluster expires in <n> days. Expiry date: <date:time> UTC. 
Please see docs.dapr.io for certificate renewal instructions to avoid service interruptions.
```

File: daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md (hunk 19 above)
Comment (msfussell) [reply daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md]: I think we should do this at the same time since it has to be done for v1.7 release anyway. The article above was updated before this new command existed and now we should ensure that the best guidance is explained. These CLI docs do not provide any guidance.

**Comment by msfussell** [comment]
@pravinpushkar - This is looking good. We just now have to address the guidance doc here and the warning message for each of the CLI commands.

File: daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md (hunk 19 above)
Comment (pravinpushkar) [reply daprdocs/content/en/reference/cli/dapr-mtls/dapr-mtls-renew-certificate.md]: @msfussell Update at the above link, The cli commands to be used for each scenario. Kindly share your thoughts on this.

**Comment by pravinpushkar** [comment]
@msfussell  Updated the PR with latest fixes related warning message and CLI commands in the operation document.

File: daprdocs/content/en/operations/security/mtls.md [hunk 21]
```diff
@@ -199,6 +199,17 @@ kubectl rollout restart statefulsets/dapr-placement-server -n <DAPR_NAMESPACE>
 
 4. Restart your Dapr applications to pick up the latest trust bundle.
 
+{{% alert title="Recommended CLI command to rotate certificates with new ones" color="success" %}}
```
Comment (msfussell) [inline daprdocs/content/en/operations/security/mtls.md]: It is hard to separate out the kubectl approach and the Dapr CLI approach here. Can you add headings for each so that these are separated? For example "Root certificate upgrade using kubectl" and "Root certificate upgrade using CLI"

File: daprdocs/content/en/operations/security/mtls.md [hunk 22]
```diff
@@ -199,6 +199,17 @@ kubectl rollout restart statefulsets/dapr-placement-server -n <DAPR_NAMESPACE>
...
+The below CLI command can be used to renew root and issuer certificates of your Kubernetes cluster. It also gives an option to provide validity for the new certificates in number of days. 
+
+The 2nd variant of this command provides `--private-key` flag to support use of `private root key` of your existing deployed certificate.
```
Comment (msfussell) [inline daprdocs/content/en/operations/security/mtls.md:205]: It is hard to separate out the kubectl approach and the Dapr CLI approach here. Can you add headings for each so that these are separated? For example "Custom root certificate upgrade using kubectl" and "Custom root certificate upgrade using CLI"

File: daprdocs/content/en/operations/security/mtls.md [hunk 23]
```diff
@@ -199,6 +199,17 @@ kubectl rollout restart statefulsets/dapr-placement-server -n <DAPR_NAMESPACE>
 
 4. Restart your Dapr applications to pick up the latest trust bundle.
 
+{{% alert title="Recommended CLI command to rotate certificates with new ones" color="success" %}}
+The below CLI command can be used to renew root and issuer certificates of your Kubernetes cluster. It also gives an option to provide validity for the new certificates in number of days.
```
Comment (msfussell) [inline daprdocs/content/en/operations/security/mtls.md]:
```suggestion
The CLI command below can be used to renew root and issuer certificates in your Kubernetes cluster. It also gives an option to provide validity for the new certificates in number of days. 
```

File: daprdocs/content/en/operations/security/mtls.md (hunk 22 above)
Comment (msfussell) [inline daprdocs/content/en/operations/security/mtls.md]:
```suggestion
The alternative variant of this command provides `--private-key` flag to support the use of a `private root key` for your existing deployed root certificate.
```

File: daprdocs/content/en/operations/security/mtls.md [hunk 24]
```diff
@@ -199,6 +199,17 @@ kubectl rollout restart statefulsets/dapr-placement-server -n <DAPR_NAMESPACE>
 
 4. Restart your Dapr applications to pick up the latest trust bundle.
 
+{{% alert title="Recommended CLI command to rotate certificates with new ones" color="success" %}}
+The below CLI command can be used to renew root and issuer certificates of your Kubernetes cluster. It also gives an option to provide validity for the new certificates in number of days. 
+
+The 2nd variant of this command provides `--private-key` flag to support use of `private root key` of your existing deployed certificate.
+This command generate new certificates signed by same private key. In this case the `Dapr Sentry service` would be able to pick the new certificate without restarting it.
```
Comment (msfussell) [inline daprdocs/content/en/operations/security/mtls.md]:
```suggestion
This command generates new certificates signed by the same private key. In this case the `Dapr Sentry service` can then read these new certificates without restarting.
```

File: daprdocs/content/en/operations/security/mtls.md [hunk 25]
```diff
@@ -246,6 +257,14 @@ Once Sentry has been completely restarted run:
 kubectl rollout restart deploy/dapr-operator -n <DAPR_NAMESPACE>
 kubectl rollout restart statefulsets/dapr-placement-server -n <DAPR_NAMESPACE>
 ```
+{{% alert title="Recommended CLI command to achieve above steps" color="success" %}}
```
Comment (msfussell) [inline daprdocs/content/en/operations/security/mtls.md]: It is hard to separate out the kubectl approach and the Dapr CLI approach here. Can you add headings for each so that these are separated? For example "Custom root certificate upgrade using kubectl" and "Custom root certificate upgrade using CLI"

File: daprdocs/content/en/operations/security/mtls.md [hunk 26]
```diff
@@ -246,6 +257,14 @@ Once Sentry has been completely restarted run:
...
+To update the provided certificates in Kubernetes cluster, the belowl CLI command can be used.
```
Comment (msfussell) [inline daprdocs/content/en/operations/security/mtls.md]:
```suggestion
To update the provided certificates in the Kubernetes cluster, the CLI command below can be used.
```

File: daprdocs/content/en/operations/security/mtls.md [hunk 27]
```diff
@@ -406,6 +425,8 @@ By default, system services will look for the credentials in `/var/run/dapr/cred
 *Note: If you signed the cert root with a different private key, restart the Dapr instances.*
 
 ## Community call video on certificate rotation
```
Comment (msfussell) [inline daprdocs/content/en/operations/security/mtls.md:479]: I realize this video is lost at the end of this article and is best move to the end of the certificate renewal section. Could you move this video up to the end of this section?

File: daprdocs/content/en/operations/security/mtls.md (hunk 21 above)
Comment (msfussell) [reply daprdocs/content/en/operations/security/mtls.md]: Also my concern here, is that given the CLI is the preferred method we should have this first BEFORE the kubectl method. So could you switch this around in the order?

File: daprdocs/content/en/operations/security/mtls.md (hunk 25 above)
Comment (msfussell) [reply daprdocs/content/en/operations/security/mtls.md]: Also my concern here, is that given the CLI is the preferred method we should have this first BEFORE the kubectl method. So could you switch this around in the order?

**Comment by msfussell** [comment]
@pravinpushkar - This is getting really close. I just left a few comments on the best practices document to walk through, mostly making sure that people understood that the CLI was preferred and that this stood out a little more with its own heading.

**Comment by msfussell** [comment]
Also review this PR https://github.com/dapr/docs/pull/2322 from Bernd.

File: daprdocs/content/en/operations/security/mtls.md (hunk 21 above)
Comment (pravinpushkar) [reply daprdocs/content/en/operations/security/mtls.md]: I am now keeping these in separate section. Kindly have a look at recent changes.

**Review by greenie-msft — APPROVED** [review]
